src/pages-and-resources/live/data/thunks.js

```diff
@@ -0,0 +1,110 @@
+import { history } from '@edx/frontend-platform';
+import { getLiveConfiguration, getLiveProviders, postLiveConfiguration } from './api';
+import {
+  DENIED, FAILED, LOADING, LOADED, SAVING, SAVED, updateStatus, updateSaveStatus,
+  updateProviders, updateAppIds, updateConfiguration,
+} from './slice';
+
+export function fetchLiveProviders(courseId) {
+  return async (dispatch) => {
+    const providers = await getLiveProviders(courseId);
+    dispatch(updateProviders(providers.providers));
+    const availableProvidersInfo = [];
+    Object.keys(providers.providers.available).forEach((key) => { availableProvidersInfo.push({ id: key }); });
+    dispatch(updateAppIds(availableProvidersInfo));
+  };
+}
+
+function updateLiveConfigurationState(config) {
+  return async (dispatch) => {
+    const data = {
+      courseKey: config?.courseKey || '',
+      enabled: config?.enabled || false,
+      consumerKey: config?.ltiConfiguration?.lti1P1ClientKey || '',
+      consumerSecret: config?.ltiConfiguration?.lti1P1ClientSecret || '',
+      launchUrl: config?.ltiConfiguration?.lti1P1LaunchUrl || '',
+      launchEmail: config?.ltiConfiguration?.ltiConfig?.additionalParameters?.customInstructorEmail || '',
+      provider: config?.providerType || 'zoom',
+      piiSharingEnable: config?.piiSharingAllowed || false,
```

Comment:
normalizeLiveConfig function is also doing  the same thing. Please try to make it `do not repeat itself`.